nft-market/components/NFT/NFT.tsx

```diff
@@ -0,0 +1,159 @@
+import Image from "next/image";
+import translations from "../../constants/en.global.json";
+import { WalletConnection } from "calimero-sdk"
+import { useState } from "react";
+import NearLogo from "../images/NearLogo";
+import { config } from "../../utils/calimeroSdk"
+
+interface Metadata {
+    description: string,
+    media: string,
+    title: string,
+}
+
+export interface NFTProps {
+    approved_accounts_ids?: object,
+    metadata: Metadata,
+    owner_id: string,
+    token_id: string,
+    sale_conditions?: string,
+    walletConnection: WalletConnection | undefined;
+    sellOrBid: boolean;
+}
+  
+export default function NFT({ approved_accounts_ids, metadata, owner_id, token_id, sale_conditions, walletConnection, sellOrBid }: NFTProps) {
+    const [sell, setSell] = useState(false);
+    const [price, setPrice] = useState("");
+    const [bid, setBid] = useState(false);
+    
+
+    const checkSellerIsOwner = () => {
+        if(!sellOrBid) {
+            if(owner_id === walletConnection?._authData.accountId) {
+                return true;
+            }
+            return false;
+        }
+    }
+
+    const handlePrice = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        setPrice(value)       
+    };
+
+    const approveNFTForSale = async (token_id: string) => {
+        let sale_conditions = {
+          sale_conditions: price,
+        };
+        if(walletConnection) {
+            await walletConnection.account().functionCall({
+                contractId: config.nftContract,
+                methodName: "nft_approve",
+                args: {
+                    token_id: token_id,
+                    account_id: config.marketContract,
+                    msg: JSON.stringify(sale_conditions),
+                },
+                gas: "300000000000000",
+                attachedDeposit: "10000000000000000000001",
+            });
+        }        
+    }
+
+    const offerPrice = async (token_id: string, price: string | undefined) => {
+        if(walletConnection) {
+            await walletConnection.account().functionCall({
```

Comment:
Make all contract calls the same way -> somewhere its walletConnection.account().functionCall({}) somewhere its contract["method_name"]()